nova_vm/src/ecmascript/types/spec/data_block.rs

```diff
@@ -61,18 +67,248 @@ mod private {
     impl Sealed for f64 {}
 }
 
-pub trait Viewable: private::Sealed {}
+pub trait Viewable: private::Sealed + Copy {
+    fn is_bigint_type() -> bool;
+    fn into_be_value(self, agent: &mut Agent) -> Value;
+    fn into_le_value(self, agent: &mut Agent) -> Value;
+    fn from_le_value(agent: &mut Agent, gc: GcScope<'_, '_>, value: Value) -> Self;
+    fn from_be_value(agent: &mut Agent, gc: GcScope<'_, '_>, value: Value) -> Self;
+}
+
+impl Viewable for u8 {
+    fn is_bigint_type() -> bool {
+        false
+    }
+
+    fn into_be_value(self, _: &mut Agent) -> Value {
+        Value::from(self.to_be())
+    }
+
+    fn into_le_value(self, _: &mut Agent) -> Value {
+        Value::from(self.to_le())
+    }
+
+    fn from_be_value(agent: &mut Agent, gc: GcScope<'_, '_>, value: Value) -> Self {
+        value.to_real(agent, gc).unwrap() as Self
```

Comment:
**suggestion:** These methods definitely call for fast-paths for integers.

Eventually we'll probably want separate methods for a "try from Value" variants that can't trigger GC.

Reply:
I just realized these should follow the conversion table of typed arrays:
https://tc39.es/ecma262/#table-the-typedarray-constructors

But yes, making these (or even the `to_*` functions) not require GC would be a very nice thing.